internal/cephfs/core/volume.go

```diff
@@ -53,20 +53,74 @@ type Subvolume struct {
 	Features   []string
 }
 
+// SubVolumeClient is the interface that holds the signature of subvolume methods
+// that interacts with CephFS subvolume API's.
+type SubVolumeClient interface {
+	// GetVolumeRootPathCeph returns the root path of the subvolume.
+	GetVolumeRootPathCeph(ctx context.Context) (string, error)
+	// CreateVolume creates a subvolume.
+	CreateVolume(ctx context.Context) error
+	// GetSubVolumeInfo returns the subvolume information.
+	GetSubVolumeInfo(ctx context.Context) (*Subvolume, error)
+	// ExpandVolume expands the volume if the requested size is greater than
+	// the subvolume size.
+	ExpandVolume(ctx context.Context, bytesQuota int64) error
+	// ResizeVolume resizes the volume.
+	ResizeVolume(ctx context.Context, bytesQuota int64) error
+	// PurgSubVolume removes the subvolume.
+	PurgeVolume(ctx context.Context, force bool) error
+
+	// CreateCloneFromSubVolume creates a clone from the subvolume.
+	CreateCloneFromSubvolume(ctx context.Context, parentvolOpt *SubVolume) error
+	// GetCloneState returns the clone state of the subvolume.
+	GetCloneState(ctx context.Context) (cephFSCloneState, error)
+	// CreateCloneFromSnapshot creates a clone from the subvolume snapshot.
+	CreateCloneFromSnapshot(ctx context.Context, snap Snapshot) error
+	// CleanupSnapshotFromSubvolume removes the snapshot from the subvolume.
+	CleanupSnapshotFromSubvolume(ctx context.Context, parentVol *SubVolume) error
+}
+
+type subVolumeClient struct {
```

Comment:
Adding comments for new types, and their attributes would be much appreciated!

Reply:
sure added it.